### ISSUE TITLE
Luaj: Multiple Expression errors for non-English locales

### ISSUE THREAD

**Issue by gnosygnu** [gnosygnu/xowa#694]
From #680
#### Repro
* Set options
  * Set `luaj` by going to `home/wiki/Special:XowaCfg?grp=xowa.addon.scribunto`
  * Set `wikitext` by going to `home/wiki/Special:XowaCfg?grp=xowa.wiki.hdumps`
* Go to `de.wikipedia.org/wiki/Bundestagswahl_2017`
  * Get bad formatting as in **Screenshot 1**
* Now, set `hdump` 
  * Article does not load; XOWA needs to be restarted to show **Screenshot 2**
* Now, set `lua`
  * Errors disappear see **Screenshot 3**

#### Screenshots
* Screenshot 1
![Screenshot_55](https://user-images.githubusercontent.com/19750226/76729150-c0f0a200-6758-11ea-806e-7bb9cd4a3d16.png)

* Screenshot 2
![Screenshot_56](https://user-images.githubusercontent.com/19750226/76729209-db2a8000-6758-11ea-8f01-424eaa4dcf96.png)

* Screenshot 3
![Screenshot_57](https://user-images.githubusercontent.com/19750226/76729283-0f9e3c00-6759-11ea-8c6e-c6b7836b6998.png)

**Comment by desb42** [comment]
The url for this page should be
`de.wikipedia.org/wiki/Bundestagswahl_2017`

**Comment by desb42** [comment]
I have just rebuilt xowa using xowa_get_and_make.sh and I cannot reproduce this
Also as a small point, dark mode seems to not show any of the graph colours
![darkmode](https://user-images.githubusercontent.com/861631/77230134-fac30d80-6b89-11ea-96d9-377ca61e8cc6.gif)

**Comment by gnosygnu** [comment]
> The url for this page should be
> de.wikipedia.org/wiki/Bundestagswahl_2017

Oops. Missed that. Corrected above. Thanks!

**Comment by gnosygnu** [comment]
Yeah, I can't reproduce it either. Tried Luaj and both hdump / wikitext. 

@Ope30: I uploaded the latest version now. Can you try installing it and seeing if it resolves the above? See https://github.com/gnosygnu/xowa/releases/tag/v4.6.6.2003

Thanks!

**Comment by Ope30** [comment]
I have tried the above mentioned repro steps. Unfortunately I always get the same result. Wikitext seems to be the problem for me. Switching to luaj will fix it, but it brings other problems with it.

But maybe it is something else. I deleted my original settings file and I still get the same results. I have changed some settings, but still get the same results. Removed read-only from all wiki files, but still the same results.


I‘m not going to give up. I'm gonna keep trying to figure out what's causing this. If I find a solution, I'll let you know immediately.

**Comment by gnosygnu** [comment]
Thanks for trying. Regarding this part:

>  Wikitext seems to be the problem for me. Switching to luaj will fix it,

In general, you should use HTML databases (instead of wikitext) and luaj (instead of lua). That said, I still can't reproduce the problem, even with lua.

So, I know you probably listed this in #680, but just asking again
* What issue makes using HTML databases so bad for 2020-02 German Wikipedia? (there's probably something, so I want to try to fix that for the next build)
* What issue makes using luaj so bad that you're using lua (there really shouldn't be anything; lua is effectively deprecated)

Thanks in advance for any info.

**Comment by Ope30** [comment]
> What issue makes using HTML databases so bad for 2020-02 German Wikipedia? (there's probably something, so I want to try to fix that for the next build)


Good question. The first reason I considered using Wikitext was that Wikitext shows you at the top (the icon, just like the original) whether the article you are reading is good/excellent. This is very useful for me. The second reason (I know it's not really a big deal) is that when using HTML, the categories in a category change positions so that they are at the top instead of at the bottom (in the original, they are at the bottom).
I think there was another reason why I considered Wikitext. In this case I'm not 100% certain, but I remember in an article that some images were not positioned correctly due to HTML. I've only been using Wikitext for a while, so I can't say much about other errors when using HTML databases.

> What issue makes using luaj so bad that you're using lua (there really shouldn't be anything; lua is effectively deprecated)

Because of


![](https://user-images.githubusercontent.com/19750226/76729150-c0f0a200-6758-11ea-806e-7bb9cd4a3d16.png)

and stuff like


![](https://user-images.githubusercontent.com/19750226/76727568-b59b7780-6754-11ea-8a82-fdaf48f1f101.png)



The change from luaj to lua or back is actually the same in the end. Both luaj and lua have problems. I will use the HTML databases as soon as these little ‘‘bugs’‘are fixed (if that is possible).

**Comment by gnosygnu** [comment]
> The first reason I considered using Wikitext was that Wikitext shows you at the top (the icon, just like the original) whether the article you are reading is good/excellent.

Ah, okay, that's a bug. Let me create an issue for that later.

> The second reason (I know it's not really a big deal) is that when using HTML, the categories in a category change positions so that they are at the top instead of at the bottom (in the original, they are at the bottom).

Ugh. Okay, that's a bug also (checked now with en.wikipedia.org/wiki/Category:Earth). I think that requires just a little extra code, but I remember there was some sort of problem with that. Will create an issue later.

> I think there was another reason why I considered Wikitext. In this case I'm not 100% certain, but I remember in an article that some images were not positioned correctly due to HTML. I've only been using Wikitext for a while, so I can't say much about other errors when using HTML databases.

There could be problems with missing images, but those should be rare. If you come across a page, just comment it here, and I'll create an issue for that.

> Because of
> and stuff like

Yeah, but the first is definitely not reproducible for me and desb42, and the second has not been reproducible for me (#688)

When you get a chance, can you try the following?
* Grab the `Generated by` from any page
  * For example, go to `home/wiki/Main_Page`
  * Scroll down to the bottom and grab the `Generated by XOWA: v4.6.5.1911 (2020-04-24 13:36:55)`
* Dump your file info for German wiki
  * Go to `home/wiki/Special:XowaDiag?type=fs.check&wiki=de.wikipedia.org`
  * Paste the results here.

I just want to confirm if we're on the same app / wiki, and figure out why there's a diff

**Comment by Ope30** [comment]
`Generated by XOWA: v4.6.5.1911 (2020-03-28 19:57:23)




scanning: E:\XOWA\XOWA 64 bit latest 9\wiki\de.wikipedia.org\
dir|data\
fil|de.wikipedia.org-core.xowa|506908672|20200217 165406.000
fil|de.wikipedia.org-file-core.xowa|879005696|20200217 170942.000
fil|de.wikipedia.org-file-ns.000-db.001.xowa|1640824832|20200217 171018.000
fil|de.wikipedia.org-file-ns.000-db.002.xowa|1638850560|20200217 171114.000
fil|de.wikipedia.org-file-ns.000-db.003.xowa|1643540480|20200217 171208.000
fil|de.wikipedia.org-file-ns.000-db.004.xowa|1645154304|20200217 171304.000
fil|de.wikipedia.org-file-ns.000-db.005.xowa|1641000960|20200217 171400.000
fil|de.wikipedia.org-file-ns.000-db.006.xowa|1640960000|20200217 171454.000
fil|de.wikipedia.org-file-ns.000-db.007.xowa|1641488384|20200217 171550.000
fil|de.wikipedia.org-file-ns.000-db.008.xowa|1643401216|20200217 171644.000
fil|de.wikipedia.org-file-ns.000-db.009.xowa|1642393600|20200217 171740.000
fil|de.wikipedia.org-file-ns.000-db.010.xowa|1645264896|20200217 171834.000
fil|de.wikipedia.org-file-ns.000-db.011.xowa|1645420544|20200217 171930.000
fil|de.wikipedia.org-file-ns.000-db.012.xowa|1648562176|20200217 172026.000
fil|de.wikipedia.org-file-ns.000-db.013.xowa|1647792128|20200217 172120.000
fil|de.wikipedia.org-file-ns.000-db.014.xowa|1646661632|20200217 172216.000
fil|de.wikipedia.org-file-ns.000-db.015.xowa|1647153152|20200217 172310.000
fil|de.wikipedia.org-file-ns.000-db.016.xowa|1646374912|20200217 172406.000
fil|de.wikipedia.org-file-ns.000-db.017.xowa|1647976448|20200217 172502.000
fil|de.wikipedia.org-file-ns.000-db.018.xowa|1653526528|20200217 172556.000
fil|de.wikipedia.org-file-ns.000-db.019.xowa|1651007488|20200217 172652.000
fil|de.wikipedia.org-file-ns.000-db.020.xowa|1662881792|20200217 172746.000
fil|de.wikipedia.org-file-ns.000-db.021.xowa|1655627776|20200217 172842.000
fil|de.wikipedia.org-file-ns.000-db.022.xowa|1670266880|20200217 172936.000
fil|de.wikipedia.org-file-ns.000-db.023.xowa|1674485760|20200217 173030.000
fil|de.wikipedia.org-file-ns.000-db.024.xowa|1674350592|20200217 173124.000
fil|de.wikipedia.org-file-ns.000-db.025.xowa|1667276800|20200217 173218.000
fil|de.wikipedia.org-file-ns.000-db.026.xowa|1664610304|20200217 173312.000
fil|de.wikipedia.org-file-ns.000-db.027.xowa|1663717376|20200217 173406.000
fil|de.wikipedia.org-file-ns.000-db.028.xowa|1655087104|20200217 173500.000
fil|de.wikipedia.org-file-ns.000-db.029.xowa|1653747712|20200217 173556.000
fil|de.wikipedia.org-file-ns.000-db.030.xowa|1655652352|20200217 173652.000
fil|de.wikipedia.org-file-ns.000-db.031.xowa|1658052608|20200217 173746.000
fil|de.wikipedia.org-file-ns.000-db.032.xowa|1652011008|20200217 173840.000
fil|de.wikipedia.org-file-ns.000-db.033.xowa|1644658688|20200217 173936.000
fil|de.wikipedia.org-file-ns.000-db.034.xowa|1640370176|20200217 174030.000
fil|de.wikipedia.org-file-ns.000-db.035.xowa|380272640|20200217 174122.000
fil|de.wikipedia.org-file-ns.004-db.001.xowa|1662844928|20200217 174138.000
fil|de.wikipedia.org-file-ns.004-db.002.xowa|1392062464|20200217 174232.000
fil|de.wikipedia.org-file-ns.014-db.001.xowa|5668864|20200217 174312.000
fil|de.wikipedia.org-file-ns.100-db.001.xowa|310431744|20200217 174314.000
fil|de.wikipedia.org-file-user.xowa|57344|20200220 093627.365
fil|de.wikipedia.org-html-ns.000-db.002.xowa|1825288192|20200217 165724.000
fil|de.wikipedia.org-html-ns.000-db.003.xowa|1876463616|20200217 165820.000
fil|de.wikipedia.org-html-ns.000-db.004.xowa|1880748032|20200217 165918.000
fil|de.wikipedia.org-html-ns.000-db.005.xowa|1876078592|20200217 170018.000
fil|de.wikipedia.org-html-ns.000-db.006.xowa|1875603456|20200217 170114.000
fil|de.wikipedia.org-html-ns.000-db.007.xowa|813019136|20200217 170210.000
fil|de.wikipedia.org-html-ns.000.xowa|1719205888|20200217 165628.000
fil|de.wikipedia.org-html-ns.004.xowa|1405378560|20200217 170236.000
fil|de.wikipedia.org-html-ns.008.xowa|618496|20200217 170326.000
fil|de.wikipedia.org-html-ns.014.xowa|195194880|20200217 170318.000
fil|de.wikipedia.org-html-ns.100.xowa|90722304|20200217 170324.000
fil|de.wikipedia.org-text-ns.000-db.002.xowa|2015633408|20200217 170514.000
fil|de.wikipedia.org-text-ns.000-db.003.xowa|2029617152|20200217 170614.000
fil|de.wikipedia.org-text-ns.000-db.004.xowa|1720573952|20200217 170714.000
fil|de.wikipedia.org-text-ns.000.xowa|1862520832|20200217 170332.000
fil|de.wikipedia.org-text-ns.004.xowa|1077395456|20200217 170426.000
fil|de.wikipedia.org-text-ns.006.xowa|53620736|20200217 170500.000
fil|de.wikipedia.org-text-ns.008.xowa|806912|20200217 170504.000
fil|de.wikipedia.org-text-ns.010.xowa|70393856|20200217 170502.000
fil|de.wikipedia.org-text-ns.012.xowa|1785856|20200217 170500.000
fil|de.wikipedia.org-text-ns.014.xowa|89915392|20200217 170506.000
fil|de.wikipedia.org-text-ns.100.xowa|52178944|20200217 170500.000
fil|de.wikipedia.org-text-ns.828.xowa|1835008|20200217 170708.000
fil|de.wikipedia.org-xtn.category.core.xowa|6266880|20200217 165526.000
fil|de.wikipedia.org-xtn.category.link-db.001.xowa|1388298240|20200217 165530.000
fil|de.wikipedia.org-xtn.search.core.xowa|189956096|20200217 165432.000
fil|de.wikipedia.org-xtn.search.link-title-ns.000-db.001.xowa|349052928|20200217 165440.000
fil|de.wikipedia.org-xtn.search.link-title-ns.999-db.001.xowa|84316160|20200217 165458.000
fil|Xowa_dewiki_2020-02_core.md5|61|20200217 165406.000
fil|Xowa_dewiki_2020-02_file_core.md5|66|20200217 170942.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.001.md5|75|20200217 171018.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.002.md5|75|20200217 171114.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.003.md5|75|20200217 171208.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.004.md5|75|20200217 171304.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.005.md5|75|20200217 171400.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.006.md5|75|20200217 171454.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.007.md5|75|20200217 171550.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.008.md5|75|20200217 171644.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.009.md5|75|20200217 171740.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.010.md5|75|20200217 171834.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.011.md5|75|20200217 171930.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.012.md5|75|20200217 172026.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.013.md5|75|20200217 172120.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.014.md5|75|20200217 172216.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.015.md5|75|20200217 172310.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.016.md5|75|20200217 172406.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.017.md5|75|20200217 172502.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.018.md5|75|20200217 172556.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.019.md5|75|20200217 172652.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.020.md5|75|20200217 172746.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.021.md5|75|20200217 172842.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.022.md5|75|20200217 172936.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.023.md5|75|20200217 173030.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.024.md5|75|20200217 173124.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.025.md5|75|20200217 173218.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.026.md5|75|20200217 173312.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.027.md5|75|20200217 173406.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.028.md5|75|20200217 173500.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.029.md5|75|20200217 173556.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.030.md5|75|20200217 173652.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.031.md5|75|20200217 173746.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.032.md5|75|20200217 173840.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.033.md5|75|20200217 173936.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.034.md5|75|20200217 174030.000
fil|Xowa_dewiki_2020-02_file_ns.000_db.035.md5|75|20200217 174122.000
fil|Xowa_dewiki_2020-02_file_ns.004_db.001.md5|75|20200217 174138.000
fil|Xowa_dewiki_2020-02_file_ns.004_db.002.md5|75|20200217 174232.000
fil|Xowa_dewiki_2020-02_file_ns.014_db.001.md5|75|20200217 174312.000
fil|Xowa_dewiki_2020-02_file_ns.100_db.001.md5|75|20200217 174314.000
fil|Xowa_dewiki_2020-02_html_ns.000.md5|68|20200217 165628.000
fil|Xowa_dewiki_2020-02_html_ns.000_db.002.md5|75|20200217 165724.000
fil|Xowa_dewiki_2020-02_html_ns.000_db.003.md5|75|20200217 165820.000
fil|Xowa_dewiki_2020-02_html_ns.000_db.004.md5|75|20200217 165918.000
fil|Xowa_dewiki_2020-02_html_ns.000_db.005.md5|75|20200217 170018.000
fil|Xowa_dewiki_2020-02_html_ns.000_db.006.md5|75|20200217 170114.000
fil|Xowa_dewiki_2020-02_html_ns.000_db.007.md5|75|20200217 170210.000
fil|Xowa_dewiki_2020-02_html_ns.004.md5|68|20200217 170236.000
fil|Xowa_dewiki_2020-02_html_ns.008.md5|68|20200217 170326.000
fil|Xowa_dewiki_2020-02_html_ns.014.md5|68|20200217 170318.000
fil|Xowa_dewiki_2020-02_html_ns.100.md5|68|20200217 170324.000
fil|Xowa_dewiki_2020-02_text_ns.000.md5|68|20200217 170332.000
fil|Xowa_dewiki_2020-02_text_ns.000_db.002.md5|75|20200217 170514.000
fil|Xowa_dewiki_2020-02_text_ns.000_db.003.md5|75|20200217 170614.000
fil|Xowa_dewiki_2020-02_text_ns.000_db.004.md5|75|20200217 170714.000
fil|Xowa_dewiki_2020-02_text_ns.004.md5|68|20200217 170426.000
fil|Xowa_dewiki_2020-02_text_ns.006.md5|68|20200217 170500.000
fil|Xowa_dewiki_2020-02_text_ns.008.md5|68|20200217 170504.000
fil|Xowa_dewiki_2020-02_text_ns.010.md5|68|20200217 170502.000
fil|Xowa_dewiki_2020-02_text_ns.012.md5|68|20200217 170500.000
fil|Xowa_dewiki_2020-02_text_ns.014.md5|68|20200217 170506.000
fil|Xowa_dewiki_2020-02_text_ns.100.md5|68|20200217 170500.000
fil|Xowa_dewiki_2020-02_text_ns.828.md5|68|20200217 170708.000
fil|Xowa_dewiki_2020-02_xtn.category.core.md5|74|20200217 165526.000
fil|Xowa_dewiki_2020-02_xtn.category.link_db.001.md5|81|20200217 165530.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.001.md5|251|20200217 170800.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.002.md5|212|20200217 170800.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.003.md5|357|20200217 170808.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.004.md5|212|20200217 170812.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.005.md5|357|20200217 170826.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.006.md5|212|20200217 170832.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.007.md5|357|20200217 170846.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.008.md5|212|20200217 170852.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.009.md5|231|20200217 170906.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.010.md5|251|20200217 170912.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.011.md5|212|20200217 170914.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.012.md5|648|20200217 170928.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.013.md5|750|20200217 170934.000
fil|Xowa_dewiki_2020-02_xtn.fulltext_search.014.md5|298|20200217 170938.000
fil|Xowa_dewiki_2020-02_xtn.search.core.md5|256|20200217 165432.000
fil|Xowa_dewiki_2020-02_xtn.search.link_title_ns.000_db.001.md5|92|20200217 165504.000
fil|Xowa_dewiki_2020-02_xtn.search.link_title_ns.999_db.001.md5|92|20200217 165522.000
fil|Xowa_dewiki_latest_files.xml|149823|20200220 090025.635
fil|Xowa_dewiki_latest_meta.sqlite|216064|20200220 090028.223

scanning: E:\XOWA\XOWA 64 bit latest 9\file\de.wikipedia.org\
dir|orig\
dir|thumb\

scanning: E:\XOWA\XOWA 64 bit latest 9\file\de.wikipedia.org\fsdb.main\

scanning: E:\XOWA\XOWA 64 bit latest 9\file\de.wikipedia.org\fsdb.user\

scanning: E:\XOWA\XOWA 64 bit latest 9\file\de.wikipedia.org\fsdb.update_00\`

**Comment by gnosygnu** [comment]
> I know that it says 4.6.5.1911, but I can assure you that I use XOWA 4.6.6.2003

Yeah, that was a bug on my side. I forgot to change the version number for the unofficial release. In this case, you have the same version as me (build timestamp is good: `(2020-03-28 19:57:23)`)

Unfortunately, I still can't reproduce the issue. I list my repro steps below. 

Before you try my repro steps, I have one random guess: 
* Go to `home/wiki/Special:XowaCfg?grp=xowa.addon.scribunto`
* Is `Enabled` checked?
  * If I uncheck it, I can reproduce your screenshot for both `luaj` (and `lua`)
  * If I leave it checked, the page shows fine: de.wikipedia.org/wiki/Bundestagswahl_2017

If `Enabled` is `checked`, then try my repro steps.

Hope this helps. Thanks.

PS:

> If possible, could you add an option to automatically extend all templates in the next version? Some are already automatically extended, for some reason not all of them. That‘d be cool!

Sorry, don't understand what this means.

---

#### Repro steps
* Go to https://github.com/gnosygnu/xowa/releases/tag/v4.6.6.2003 and download `xowa_app_windows_64_v4.6.6.2003.zip`
* Unzip it to a temp directory like `C:\xowa_temp`
* Move your German wiki folder to `C:\xowa_temp` 
  * Basically, you want a folder like `C:\xowa_temp\wiki\de.wikipedia.org`
  * (REMEMBER TO MOVE IT BACK TO WHEREVER IT WAS WHEN DONE)
* Launch XOWA through `xowa_64.exe`
* Go to `home/wiki/Special:XowaCfg?grp=xowa.wiki.hdumps`
  * Uncheck `Prefer HTML Databases for Read tab`
* Go to `home/wiki/Special:XowaCfg?grp=xowa.addon.scribunto`
   * Make sure `Enabled` is checked
   * Make sure `Lua engine` is `luaj`
* Go to `de.wikipedia.org/wiki/Bundestagswahl_2017`
* You should see no `Expression errors`

**Comment by Ope30** [comment]
`Enabled` was always checked.



> You should see no Expression errors

I wish that was the case, because they're still there. I did exactly what you told me, but for some damn reason they're just still there. :(


I have downloaded and installed Java Development Kit 14.0.1, but no changes. I even tried to use older versions of XOWA, but it's always the same error. I'm starting to think that I might be missing some files, have you compared my files with yours? German Wikipedia is 81.6GB in size.


Probably the best solution for me is to use HTML all the time and not Wikitext once and for all. As I said before, there were only a few reasons why I decided to use Wikitext. To be honest, I miss the speed of HTML so much.

**Comment by gnosygnu** [comment]
> Enabled was always checked.

Yeah, that's disappointing. How about going here?: home/wiki/Diagnostics/Scribunto/All

The left column should show "pass" in green. Any "fail" in red would be a problem

Again, just make sure that the following is set

```
Go to home/wiki/Special:XowaCfg?grp=xowa.addon.scribunto
* Make sure Enabled is checked
* Make sure Lua engine is luaj
```

**Comment by gnosygnu** [comment]
Fixed with https://github.com/gnosygnu/xowa/releases/tag/v4.6.6.2005. Give it a try; hopefully it works! Also, this may fix #680 as well

Thanks again @desb42!

---

As per @desb42's comment in #717, this issue is related to LuaJ using the default user Locale. Luaj should insead use a US locale (that's effectively what the C version does, which is locale-unaware)

Was able to reproduce on my side by:
* On my Windows box, changed `Region and Language` from `United States` to `Germany`
* Navigated to `home/wiki/Diagnostics/Scribunto/All` and seeing a few failures for `formatnumber`
* Navigated to `de.wikipedia.org/wiki/Bundestagswahl_2017` and seeing multiple failures similar to the screenshot above.
* Installed 4.6.6.2005
  * Above pages no longer have errors

**Comment by Ope30** [comment]
> Fixed with https://github.com/gnosygnu/xowa/releases/tag/v4.6.6.2005. Give it a try; hopefully it works! Also, this may fix #680 as well

NICE! I can confirm that the issues with lua and luaj have been solved! Once again, you and @desb42 have done an amazing job. Thank you both for taking the time to fix this! I'm so glad that this is over now. I've always wondered why I was the only one who came across these particular issues. The explanation makes sense. I myself would never have thought that it had anything to do with my national PC language. I guess one had to dig deep in the shit to find that out.


I hate to say this, but I think there is something wrong with full text search because it doesn't seem to work for me anymore. :(

**Comment by gnosygnu** [comment]
> NICE! I can confirm that the issues with lua and luaj have been solved! Once again, you and @desb42 have done an amazing job. Thank you both for taking the time to fix this!

All credit to @desb42! He was the one who noticed the German menus. Kudos to him for that sharp eye in noticing all the details! :clap: :clap: :clap:

> I'm so glad that this is over now. I've always wondered why I was the only one who came across these particular issues. The explanation makes sense. I myself would never have thought that it had anything to do with my national PC language. I guess one had to dig deep in the shit to find that out.

Also, thanks to you for sticking with it! This was a significant Luaj bug for non-US locales, and I'm glad to have it fixed!

> I hate to say this, but I think there is something wrong with full text search because it doesn't seem to work for me anymore. :(

Ugh. Sorry to hear. Based on https://github.com/gnosygnu/xowa/issues/694#issuecomment-625332758 it looks like you have anything in `E:\XOWA\XOWA 64 bit latest 9\wiki\de.wikipedia.org\data\search\java7-v1`.

If you don't have data there, can you try rebuilding as per instructions here: `home/wiki/App/Full-text_search/Lucene/Search_indexes/Building`

**Comment by Ope30** [comment]
MY BAD! There is NOTHING wrong with the latest build. The reason was a newer Java version (Java Development Kit 14.0.1), which I installed and replaced over an older version (8.0.1010.13). Everything is fine now!

**Comment by gnosygnu** [comment]
> MY BAD! There is NOTHING wrong with the latest build. The reason was a newer Java version (Java Development Kit 14.0.1), which I installed and replaced over an older version (8.0.1010.13). Installed 8.0.1010.13 again. Full text search is working fine now!

Cool! Good to hear!

Created issues for the other 2 bugs above. Otherwise planning to close this specific one. Thanks!

**Comment by Ope30** [comment]
> Created issues for the other 2 bugs above. Otherwise planning to close this specific one. Thanks!

Okay. 👍🏼
